### PR TITLE
[MM-61106] fix server/channels/app/desktop_login.go errcheck issues

### PR DESCRIPTION
**Pull request by kyrillosisaac2**
**Summary**
Fix errcheck issues in server/channels/app/desktop_login.go

**Ticket Link**
Fixes https://github.com/mattermost/mattermost/issues/28812
https://mattermost.atlassian.net/browse/MM-61106

```release-note
NONE
```

### CODE REVIEW

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @kyrillosisaac2!

File: server/channels/app/desktop_login.go [hunk 1]
```diff
@@ -14,8 +14,9 @@ func (a *App) GenerateAndSaveDesktopToken(createAt int64, user *model.User) (*st
 	err := a.Srv().Store().DesktopTokens().Insert(token, createAt, user.Id)
 	if err != nil {
 		// Delete any other related tokens if there's an error
-		a.Srv().Store().DesktopTokens().DeleteByUserId(user.Id)
-
+		if deleteErr := a.Srv().Store().DesktopTokens().DeleteByUserId(user.Id); deleteErr != nil {
+			return nil, model.NewAppError("GenerateAndSaveDesktopToken", "app.desktop_token.delete.error", nil, "", http.StatusInternalServerError).Wrap(deleteErr)
+		}
```
Comment (hanzei) [inline server/channels/app/desktop_login.go:19]: By returning an error here, we obfuscate the error returned by `a.Srv().Store().DesktopTokens().Insert`. I think we should log the error instead. 

What do you think @devinbinnie ?

Comment (devinbinnie) [reply server/channels/app/desktop_login.go:19]: Sorry I must have misread this, yeah we should really be returning the `Insert` error here instead, as that's the real issue

Comment (kyrillosisaac2) [reply server/channels/app/desktop_login.go:19]: I will fix it ASAP

File: server/channels/app/desktop_login.go [hunk 2]
```diff
@@ -27,22 +29,26 @@ func (a *App) ValidateDesktopToken(token string, expiryTime int64) (*model.User,
 	userId, err := a.Srv().Store().DesktopTokens().GetUserId(token, expiryTime)
 	if err != nil {
 		// Delete the token if it is expired or invalid
-		a.Srv().Store().DesktopTokens().Delete(token)
-
+		if deleteErr := a.Srv().Store().DesktopTokens().Delete(token); deleteErr != nil {
+			return nil, model.NewAppError("ValidateDesktopToken", "app.desktop_token.delete.error", nil, "", http.StatusInternalServerError).Wrap(deleteErr)
```
Comment (devinbinnie) [inline server/channels/app/desktop_login.go:33]: This one is returning the delete error as well. Can we fix it for all of these?

**Review by devinbinnie — CHANGES_REQUESTED** [review]

**Review by devinbinnie — APPROVED** [review]
Thanks @kyrillosisaac2, LGTM!

**Review by hanzei — APPROVED** [review]

**Pull request merged**